### PR TITLE
add nodeSelector and Affinity to Cluster controller

### PR DESCRIPTION
**Pull request by SmaineTF1**
## What does this PR change?
Add `nodeSelector` and `affinity` to the cluster controller.

## Does this PR rely on any other PRs?
No

## How does this PR impact users? (This is the kind of thing that goes in release notes!)
Add `nodeSelector` and `affinity` to the cluster controller.

## Links to Issues or tickets this PR addresses or fixes
None

## What risks are associated with merging this PR? What is required to fully test this PR?
None

## How was this PR tested?
With a `helm template` command and the following values

```yaml
clusterController:
  enabled: true
  image:
    repository: public.ecr.aws/kubecost/cluster-controller
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
          - matchExpressions:
              - key: kubernetes.io/instance-type
                operator: In
                values:
                  - m5.large
  tolerations:
    - key: dedicated
      operator: Equal
      value: cost-analyzer
      effect: NoSchedule
  nodeSelector:
    kubernetes.io/instance-type: m5.large
 ```

## Have you made an update to documentation? If so, please provide the corresponding PR.
No

### CODE REVIEW

File: cost-analyzer/values-eks-cost-monitoring.yaml [hunk 1]
```diff
@@ -40,4 +40,3 @@ prometheus:
 
 reporting:
   productAnalytics: false
-
```
Comment (enzo-ctosolutions) [inline cost-analyzer/values-eks-cost-monitoring.yaml]: Can you add a new line pls

**Review by enzo-ctosolutions — APPROVED** [review]

**Comment by jessegoodier** [comment]
@SmaineTF1 thanks for this. Could you also add the corresponding nulls to the values?
https://github.com/kubecost/cost-analyzer-helm-chart/blob/667910da8c09ba48b039fadac20fa76a3fab482f/cost-analyzer/values.yaml#L2744

**Comment by chipzoller** [comment]
Bump @SmaineTF1

**Comment by SmaineTF1** [comment]
Sorry for the delay, I just added the default values.

**Review by JesseGoodier — APPROVED** [review]
tested with and without the new vales

**Pull request merged**